pds/server.go

```diff
@@ -265,7 +274,19 @@ func (s *Server) readRecordFunc(ctx context.Context, user bsutil.Uid, c cid.Cid)
 	return lexutil.CborDecodeValue(blk.RawData())
 }
 
-func (s *Server) RunAPI(listen string) error {
+func (s *Server) RunAPI(addr string) error {
+	var lc net.ListenConfig
+	ctx, cancel := context.WithTimeout(context.Background(), serverListenerBootTimeout)
+	defer cancel()
+
+	li, err := lc.Listen(ctx, "tcp", addr)
```

Comment:
just to confirm, since the docs arent super clear without diving into the implementation, the lifetime of this context only governs the creation of the listener and allocation of the file descriptor, right? Its not going to break the listener once the timeout occurs, is it?

Reply:
Correct! I've used this API successfully in other cases but also I traced through all the code (through [internetSocket](https://github.com/golang/go/blob/dbf9bf2c39116f1330002ebba8f8870b96645d87/src/net/ipsock_posix.go#L137), and [netFD#dial](https://github.com/golang/go/blob/dbf9bf2c39116f1330002ebba8f8870b96645d87/src/net/sock_posix.go#L116) and such) to double-check. And, yep, it's just for the boot time. I might post them a PR to add to the docs.